api/src/infrastructure/postgres-survey-answers.repository.ts

```diff
@@ -70,71 +69,80 @@ export class PostgresSurveyAnswerRepository
 
   public async addSurveyDataToBaseWidget(
     widget: BaseWidgetWithData,
-    filters?: WidgetDataFilters,
+    params: { filters?: WidgetDataFilter[]; breakdown?: string } = {},
   ): Promise<BaseWidgetWithData> {
+    const { filters, breakdown } = params;
     let filterClause: string;
     if (filters !== undefined) {
       filterClause = this.sqlAdapter.generateSqlFromWidgetDataFilters(filters);
     }
-
-    await this.appendBaseWidgetData(widget, filterClause);
+    if (breakdown === undefined) {
+      await this.addDataToWidget(widget, filterClause);
+    } else {
+      await this.addBreakdownDataToWidget(widget, breakdown, filterClause);
+    }
     return widget;
   }
 
-  private async appendBaseWidgetData(
+  private async addDataToWidget(
     widget: BaseWidgetWithData,
     filterClause: string,
   ): Promise<void> {
-    const { indicator } = widget;
+    widget.data = {};
 
     // Check if the indicator is an edge case
-    const methodName = this.edgeCasesMethodNameMap[indicator];
+    const methodName = this.edgeCasesMethodNameMap[widget.indicator];
     if (methodName !== undefined) {
       return this[methodName](widget, filterClause);
     }
 
-    const widgetData: WidgetData = {};
     const [
       supportsChart,
-      supportsSingleValue,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      supportsSingleValue, // No generic implementation for single value widgets for the time being
       supportsMap,
-      supportsNavigation,
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      supportsNavigation, //  No generic implementation navigation widgets for the time being
```

Comment:
Nit: Maybe put these comments in `getSupportedVisualizations` method, so you don't have to use the eslint ignore rule?